fix: default head requests for endpoints when no explicit head method… (#13210)
withastro/astro: .changeset/afraid-turkeys-kneel.md

---
'astro': minor
---

> Provide default implementation for `HEAD` requests for endpoints which not explicitly implemented it
> 
> If an endpoint defines a `GET` handler but no specific handler for `HEAD`, Astro will now call the `GET` handler and return the headers and status but an empty body. 
> If you would like to have custom handling for `HEAD` requests you can define a handler for it and it will use that instead.

Reviewer: Changelog suggestion:
Removed section about custom handling, since this behaviour is used when a handler is not defined. Also rewrote the opening sentence and simplified the second.
```suggestion
Handle `HEAD` requests to an endpoint when a handler is not defined.

If an endpoint defines a handler for `GET`, but does not define a handler for `HEAD`, Astro will call the `GET` handler and return the headers and status but an empty body.
```
